R/approx-cdf.R

```diff
@@ -144,11 +145,16 @@ approx_cdf_from_quantiles <- function(quantiles, probs) {
   assert_that(!is.unsorted(quantiles))
   assert_that(is_double(probs))
   assert_that(all(0 <= probs & probs <= 1))
-  ##
+
+  # NOTE: What's the purpose of this run-length encoding?
   rle.quantiles <- rle(quantiles)
+  # NOTE: cdf.xs appears to be identical to quantiles.
   cdf.xs <- rle.quantiles[["values"]]
+  # NOTE: cdf.ys.at.inds appears to be identical to seq_along(probs).
   cdf.ys.at.inds <- cumsum(rle.quantiles[["lengths"]])
   cdf.ys.before.inds <- cdf.ys.at.inds - rle.quantiles[["lengths"]] + 1L
+  # NOTE: The following two appear to be identical to c(0.0, probs, 1.0), but
+  # shifted by 1 index.
```

Comment:
```suggestion
```

They should be overriding stuff, not shifting or appending.  This is adding the point masses at the lowest and highest estimated quantiles (might call this "point mass extrapolation").